### ISSUE TITLE
Cities Importance 

### ISSUE THREAD

**Issue by MiklerGM** [chronhq/backend#78]
Problem: capitals, cities, towns and villages should became visible on different zoom levels.

Find the best approach to filter cities by it's importance for best UI experience.
On frontend filtering will be done via MVT filtering options https://www.mapbox.com/mapbox-gl-js/style-spec/#other-filter

Mapbox will automatically hide collided items, so we don't need clustering algorithm.

The proposal: add one or more fields to determine city importance for current period of time

> We could add additional field in the `api_city` table named `scale_rank` and set it depending on the `instance of` of wikidata item. 
> If we bind scale rank to zoom we could do it like this
> 
> scale rank | zoom | cities
> --- | --- | ---
> 1 | >= 1 | capitals or former capitals
> 2 | >= 2 | regional/state capitals
> 3 | >= 3 | two major cities of the country (not including [1])
> 5 | >= 5 | four major cities of the country (not including [1, 3])
> 7 | >= 7 | six largest cities (not including [1, 3, 5])
> ... | ... | ...
> 
> In this case for zoom level 7 we will have `1 + (?)+ 2 + 4 + 6 = 13` cities (or more) per country.
> 
> It would be easy to apply filters like this on the frontend
> 
> The problem that will remain, scale rank of city can be changed during it's existence
> Some countries are small and 3 major cities can be really close to each other to show them on the high level zooms
> 

_Originally posted by @MiklerGM in https://github.com/chronhq/backend/pull/70#issuecomment-463981481_

**Comment by ataalik** [comment]
is there a field on wikidata that inherently gives us the size of the settlement, ie cities, villages, towns

**Comment by whirish** [comment]
@ataalik There's a few but I'm not sure how standardized they are. https://www.wikidata.org/wiki/Q1049415 is an instance of "village". https://www.wikidata.org/wiki/Q84 is an instance of "capital". https://www.wikidata.org/wiki/Q2256 is an instance of "metropolis". https://www.wikidata.org/wiki/Q189288 is an instance of "town", and so on. A lot of semi-major cities seem to just be an instance of "city" and nothing else, which might make this a bit difficult. We can also ranking cities from a service other than wikidata perhaps.

**Comment by MiklerGM** [comment]
> is there a field on wikidata that inherently gives us the size of the settlement, ie cities, villages, towns

We are working with historical data, town in 1000 A.D. might be more important than metropolis in 2019 A.D.
As an example, Edo was renamed to Tokyo and capital was moved there, and before that city was less important than Kyoto at some point.

> We can also ranking cities from a service other than wikidata perhaps.

Yeah, there was some open database from national geographic with `scaleRank` we should ask @v1kun about it.
In chronist we were using two fields scale rank + city class

**Comment by v1kun** [comment]
I agree with @MiklerGM, it' pretty difficult to manage importance, I believe some of this should be done manualy.

I borrowed scaleRank from [http://www.naturalearthdata.com/](http://www.naturalearthdata.com/) database (not national geographic 😃). It's suitable, but anyway back-in-time mapping is difficult for this matter anyway.

**Comment by v1kun** [comment]
It's also big problem about wikidata's 'instance of', I'm not sure how to manage this issue.

**Comment by ataalik** [comment]
I doubt it exists but if we had historical census data from settlements we could somehow calculate importance using stuff like population, the year the data is from and proximity to other cities.

**Comment by MiklerGM** [comment]
We got population for 20th century, not more.
We could track number of cross links from events for current period?
